docs/reference/storage.md

```diff
@@ -0,0 +1,116 @@
+# Storage
+
+uv persists data in several locations on your system.
+
+## Directory Strategies
+
+uv follows platform conventions (like
+[XDG](https://specifications.freedesktop.org/basedir-spec/latest/) on Unix) for determining where to
+store different types of data. Generally, it's best to configure these rather than each uv-specific
+storage location.
```

Comment:
```suggestion
For determining where to store different types of data, uv follows the [XDG](https://specifications.freedesktop.org/basedir-spec/latest/) conventions on Linux and macOS and the platform defaults on Windows. Generally, it's best to configure these rather than each uv-specific storage location.
```

It kind of an awkward sentence cause XDG is kinda the platform default on Linux, expect it's not defined by Linux but something people aligned on, we're using XDG on macOS over the macOS platform APIs because we want to treat it more like a Unix that follows XDG, which aligns better with dev usage, and then we're following the actual platform conventions on Windows (but we also had a thread about respecting XDG on Windows when set, and falling back to the native APIs).